src/main/java/org/zalando/nakadi/controller/PostSubscriptionController.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.web.util.UriComponents;
+import org.zalando.nakadi.domain.Subscription;
+import org.zalando.nakadi.domain.SubscriptionBase;
+import org.zalando.nakadi.exceptions.runtime.DuplicatedSubscriptionException;
+import org.zalando.nakadi.exceptions.runtime.InconsistentStateException;
+import org.zalando.nakadi.exceptions.runtime.MyNakadiRuntimeException1;
+import org.zalando.nakadi.exceptions.runtime.NoEventTypeException;
+import org.zalando.nakadi.exceptions.runtime.NoSubscriptionException;
+import org.zalando.nakadi.exceptions.runtime.TooManyPartitionsException;
+import org.zalando.nakadi.exceptions.runtime.WrongInitialCursorsException;
+import org.zalando.nakadi.plugin.api.ApplicationService;
+import org.zalando.nakadi.problem.ValidationProblem;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.subscription.SubscriptionService;
+import org.zalando.nakadi.util.FeatureToggleService;
+import org.zalando.problem.MoreStatus;
+import org.zalando.problem.Problem;
+import org.zalando.problem.spring.web.advice.Responses;
+
+import javax.validation.Valid;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import static org.springframework.http.HttpStatus.NOT_IMPLEMENTED;
+import static org.springframework.http.HttpStatus.OK;
+import static org.zalando.nakadi.util.FeatureToggleService.Feature.CHECK_OWNING_APPLICATION;
+import static org.zalando.nakadi.util.FeatureToggleService.Feature.DISABLE_SUBSCRIPTION_CREATION;
+import static org.zalando.nakadi.util.FeatureToggleService.Feature.HIGH_LEVEL_API;
+
+
+@RestController
+public class PostSubscriptionController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PostSubscriptionController.class);
+
+    private final FeatureToggleService featureToggleService;
+    private final ApplicationService applicationService;
+    private final SubscriptionService subscriptionService;
+
+    @Autowired
+    public PostSubscriptionController(final FeatureToggleService featureToggleService,
+                                      final ApplicationService applicationService,
+                                      final SubscriptionService subscriptionService) {
+        this.featureToggleService = featureToggleService;
+        this.applicationService = applicationService;
+        this.subscriptionService = subscriptionService;
+    }
+
+    @RequestMapping(value = "/subscriptions", method = RequestMethod.POST)
+    public ResponseEntity<?> createOrGetSubscription(@Valid @RequestBody final SubscriptionBase subscriptionBase,
+                                                     final Errors errors,
+                                                     final NativeWebRequest request,
+                                                     final Client client) {
+        if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
+            return new ResponseEntity<>(NOT_IMPLEMENTED);
```

Comment:
Yes, we still need that. We are considering temporary switching off HILA when migrating new new versions of cursors and when decoupling zookeepers.